### ISSUE TITLE
Delete via firebase firestore:delete no longer works

### ISSUE THREAD

**Issue by ghost** [firebase/firebase-tools#521]
### Version info

    "firebase": "^4.6.0",
    "firebase-tools": "^3.13.1",

### Steps to reproduce

`firebase firestore:delete -r -y users/test/comments`

### Expected behavior

Successful delete

### Actual behavior
Error: An unexpected error has occurred.

firebase-debug.log:

```
[debug] [] > authorizing via signed-in user
[debug] [] >>> HTTP REQUEST GET https://admin.firebase.com/v1/projects/firestore-[redacted]
[debug] [] <<< HTTP RESPONSE 200
[debug] [] >>> HTTP REQUEST GET https://admin.firebase.com/v1/database/firestore-[redacted]/tokens
[debug] [] <<< HTTP RESPONSE 200
[debug] [] >>> HTTP REQUEST POST https://firestore.googleapis.com/v1beta1/projects/firestore-[redacted]/databases/(default)/documents:commit { writes:
   [ { delete: 'projects/firestore-[redacted]/databases/(default)/documents/[redacted]' } ] }
[debug] [] <<< HTTP RESPONSE 400
[debug] [] <<< HTTP RESPONSE BODY
[debug] [] deletePath:initialDelete:error
[debug] [] TypeError: Cannot read property 'allDescendants' of undefined
    at /usr/local/lib/node_modules/firebase-tools/lib/firestore/delete.js:295:18
    at tryCatch (/usr/local/lib/node_modules/firebase-tools/node_modules/rsvp/dist/rsvp.js:525:12)
    at invokeCallback (/usr/local/lib/node_modules/firebase-tools/node_modules/rsvp/dist/rsvp.js:538:13)
    at publish (/usr/local/lib/node_modules/firebase-tools/node_modules/rsvp/dist/rsvp.js:508:7)
    at publishRejection (/usr/local/lib/node_modules/firebase-tools/node_modules/rsvp/dist/rsvp.js:443:3)
    at flush (/usr/local/lib/node_modules/firebase-tools/node_modules/rsvp/dist/rsvp.js:2415:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
[error]
[error] Error: An unexpected error has occurred.
```

**Comment by merlinnot** [comment]
It works for documents, it does not work for collections. As a workaround, you can create a file which contains a list of paths to documents (each in a new line), like this:
```
users/test/comments/[id_1]
users/test/comments/[id_2]
users/test/comments/[id_3]
...
```
and run
```bash
while read in; do firebase firestore:delete -r -y "$in"; done < [FILE_NAME]
```
for example
```bash
while read in; do firebase firestore:delete -r -y "$in"; done < list_of_comments.txt
```

**Comment by notsonotso** [comment]
@merlinnot it used to work fine to delete collections recursively.

**Comment by merlinnot** [comment]
That's most definitely a serious issue, for me too. I just gave you a workaround till it will be patched.

**Comment by samtstern** [comment]
Apologies all I did not see this issue!  I will get on this ASAP.

**Comment by samtstern** [comment]
@ghost I believe part of this is fixed by some things already released (your issue is filed against an older version) and the rest is fixed by #541

**Comment by samtstern** [comment]
Closed via #541
